adapters/applogy/applogy.go

```diff
@@ -45,6 +45,7 @@ func (a *ApplogyAdapter) MakeRequests(request *openrtb.BidRequest, _ *adapters.E
 				banner.W = &format.W
 				banner.H = &format.H
 			}
+			impression.Banner = &banner
```

Comment:
You should move the banner copy and reference inside the `if banner.W == nil || banner.H == nil || *banner.W == 0 || *banner.H == 0 {` statement, because otherwise the copy isn't necessary.

Reply:
Also, `impression` is a copy of an element in the `Imp` slice. The assignment modifies the the `Banner` pointer of the copy, not the actual impression within the `Imp` array. The `Imp` slice is already a shallow copy of the original request, so it is safe to modify the `Banner` pointer in one of the elements.

Reply:
Ah, that's a good point @hhhjort. It shouldn't have failed the new memory tests then though?

Reply:
Did we throw a false positive here then?

Reply:
It doesn't modify the existing Banner (which would be bad as that is pointing to the shared original request Banner). In line 36 (banner := *impression.Banner), you make `banner` a local shallow copy of the original `Banner`. You then modify that local copy. Line 48 you assign `impression.Banner`, where `impression` is the loop copy of the actual element in `Imp` to point to the new local copy of `banner`. So the temporary `impression` imp object gets it's `Banner` pointer assigned to a new, local `banner` object, and the object in the `Imp` slice that we are iterating over is unchanged. Thus when the loop ends, `request.Imp[n].Banner` is unchanged.

Reply:
I committed an update, please let me know if we got it right this time